framework/src/modules/interoperability/mainchain/commands/recover_message.ts

```diff
@@ -162,6 +163,14 @@ export class RecoverMessageCommand extends BaseInteroperabilityCommand<Mainchain
 					error: new Error('Cross-chain message sending chain is not live.'),
 				};
 			}
+
+			// Check that sending chain is valid. This check is relevant only in case direct sidechain channels are enabled
+			if (!chainID.equals(getMainchainID(chainID)) && !ccm.sendingChainID.equals(chainID)) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Cross-chain message sending chain ID is not valid.'),
+				};
+			}
```

Comment:
This check is not needed in the mainchain. As the comment in the LIP says 

```
        # Check that sending chain is valid. This check is relevant only in case direct sidechain channels are enabled
        if OWN_CHAIN_ID != getMainchainID() and ccm.sendingChainID != OWN_CHAIN_ID:
            raise  Exception("Cross-chain message sending chain ID is not valid.")
```

Currently (no direct channels) the message recovery command exists only in mainchain and this extra check is not needed. It is anyways pointless since the `OWN_CHAIN_ID != getMainchainID()` would never be true in mainchain, so adding it would never change the outcome of the verification.

In the future, in case direct sidechain channels get enabled, then the message recovery command would also need to be supported in the sidechains. There, we would need this check to ensure the validity (this check essentially says that sidechains can never forward messages from other chains). 

So this check would be needed only in case message recovery gets enabled in sidechains.